File: Clojure/rdm.clj
;; File: rdm.clj
;; Author: Collin J. Doering <rekahsoft@gmail.com>
;; Description: A rdm file for playing around with

(map (fn [x] (* x x)) '(0 1 2 3 4 5))

;; Example factorial function with accumulator (tail call)
(defn factorial
  ([n]
     (factorial n 1))
  ([n acc]
     (if  (= n 0) acc
          (recur (dec n) (* acc n)))))

File: Ruby/rdm.rb
# File: rdm.rb
# Date: 02/10/2010
# Author: Collin J. Doering <rekahsoft@gmail.com>
# Description: Random source file to experiment while learning ruby

class Point
  attr_accessor :x, :y

  def initialize(x=0,y=0)
    @x, @y = x, y
  end

  def self.add(p1,p2)
    Point.new(p1.x + p2.x, p1.y + p2.y)
  end
 end
